patches/echarts+5.5.0.patch

```diff
@@ -11,23 +25,23 @@ index fa02e2f..cc3cb7d 100644
 +  let stackResultValue = NaN;
    if (dataCoordInfo.stacked) {
 -    value = data.get(data.getCalculationInfo('stackedOverDimension'), idx);
--  }
--  if (isNaN(value)) {
--    value = dataCoordInfo.valueStart;
 +    stackedOverValue = data.get(data.getCalculationInfo('stackedOverDimension'), idx);
 +    stackResultValue = data.get(data.getCalculationInfo('stackResultDimension'), idx);
++  } else {
++    stackResultValue = data.get(dataCoordInfo.valueDim, idx);
    }
-+  if (isNaN(stackedOverValue) && !(dataCoordInfo.stacked && isNaN(stackResultValue))) {
+-  if (isNaN(value)) {
+-    value = dataCoordInfo.valueStart;
++  if (isNaN(stackedOverValue) && !isNaN(stackResultValue)) {
```

Comment:
This `getStackedOnPoint` for a given point of series computes the bottom point of the area polygon which is forms the bottom of the area polygon. Here it is called `stackedOverValue`. On stacked charts the bottom line should be placed on the previous series in stack and on non-stacked charts it should be the areaStyle.origin which is 0 by default. However, if a series has no value for a given x value, then we do not need to have this `stackedOverValue`:
<img width="165" alt="Screenshot 2024-06-12 at 9 06 50 PM" src="https://github.com/metabase/metabase/assets/14301985/5d2d2586-1a56-4c77-b4fa-47c294a26a61">